src/states/common/base.rs

```diff
@@ -62,15 +65,73 @@ pub trait Base: Display {
         };
     }
 
-    // Sends the given `bytes` to the peer with the given `dst_id`. If that results in an
+    // Sends the given `data` to the peer with the given `dst_id`. If that results in an
     // error, it disconnects from the peer.
-    fn send_or_drop(&mut self, dst_id: &PublicId, bytes: Vec<u8>, priority: u8) {
-        if let Err(err) = self.crust_service().send(dst_id, bytes, priority) {
+    fn send_or_drop(&mut self, dst_id: &PublicId, data: CrustSendReceiveData, priority: u8) {
+        if let Err(err) = self.crust_service().send(dst_id, data, priority) {
             info!("{} Connection to {} failed: {:?}", self, dst_id, err);
             // TODO: Handle lost peer, but avoid a cascade of sending messages and handling more
             //       lost peers: https://maidsafe.atlassian.net/browse/MAID-1924
             // self.crust_service().disconnect(*pub_id);
             // return self.handle_lost_peer(*pub_id).map(|_| Err(err.into()));
         }
     }
+
+    fn to_curst_send_receive_data(
+        &self,
+        message: Message,
+    ) -> Result<CrustSendReceiveData, (serialisation::SerialisationError, Message)> {
+        to_curst_send_receive_data(message)
+    }
+
+    fn from_curst_send_receive_data(
+        &self,
+        data: CrustSendReceiveData,
+    ) -> Result<Message, RoutingError> {
+        from_curst_send_receive_data(data)
+    }
+
+    // Serialise HopMessage containing the given signed message.
+    fn to_hop_bytes(
+        &self,
+        signed_msg: SignedMessage,
+        route: u8,
+        sent_to: BTreeSet<XorName>,
+    ) -> Result<CrustSendReceiveData, RoutingError> {
+        let hop_msg = HopMessage::new(
+            signed_msg,
+            route,
+            sent_to,
+            self.full_id().signing_private_key(),
+        )?;
+        let message = Message::Hop(hop_msg);
+        Ok(self
+            .to_curst_send_receive_data(message)
+            .map_err(|(err, _)| err)?)
+    }
+}
+
+#[cfg(not(feature = "mock_serialise_send"))]
+fn to_curst_send_receive_data(
+    message: Message,
+) -> Result<CrustSendReceiveData, (serialisation::SerialisationError, Message)> {
+    let result = serialisation::serialise(&message);
+    result.map_err(|err| (err, message))
+}
+
+#[cfg(not(feature = "mock_serialise_send"))]
+fn from_curst_send_receive_data(data: CrustSendReceiveData) -> Result<Message, RoutingError> {
+    serialisation::deserialise(&data).map_err(RoutingError::SerialisationError)
+}
+
+#[cfg(feature = "mock_serialise_send")]
+fn to_curst_send_receive_data(
+    message: Message,
+) -> Result<CrustSendReceiveData, (serialisation::SerialisationError, Message)> {
+    Ok(message)
+}
+
+#[cfg(feature = "mock_serialise_send")]
+fn from_curst_send_receive_data(data: CrustSendReceiveData) -> Result<Message, RoutingError> {
```

Comment:
s/curst/crust (Don't mean to be noisy, but I thought it may help to have all instances pointed out for checking after you've renamed them all)